Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -128,7 +134,7 @@ - (void)beginLocationUpdates
 {
   if (!_locationManager) {
     _locationManager = [CLLocationManager new];
-    _locationManager.distanceFilter = RCT_DEFAULT_LOCATION_ACCURACY;
+    _locationManager.distanceFilter = _observerOptions.distanceFilter;
```

Comment:
Nevermind there is a fix https://github.com/facebook/react-native/commit/49b2805da2237c60559b779f3f6475914f311698